### DISCUSSION TITLE
Welcome to Ram-Tester Discussions!

### DISCUSSION THREAD

**Discussion by tops4u** [tops4u/Ram-Tester#1] [Announcements]
## 👋 Welcome!
 Feel free to ask questions or suggestions for this little project.

**Comment by glaszczkowski** [comment]
Hi, thank you for the project!. I have noticed that there are different values of R1 and R6 on the schematic (TH) and in the BOM. What are the correct values?

**Reply by tops4u** [reply]
Hi, 

thanks for pointing this out. I missed this due to some redesign with the FET driving the LED, those resistors are just current limiters for the LED. However both values will work and not fry your LED. Probably the values from the BOM are a better as the LED will not get super bright. Any value from 100 to around 750 Ohms will work well and just dim the LED more or less.

**Reply by glaszczkowski** [reply]
Yes, I have assembled the tester, and had to use even higher value of the R1 resistor as the green almost burnt my eyes ;) It depends of the diode of course. However this is a great and useful project. Thank you :-)

**Comment by glaszczkowski** [comment]
Hi, do you plan to publish the v2 version of TH board (OLED) on pcbway?

**Reply by tops4u** [reply]
Hi, I still need a "Guinea Pig" to test the new Design ;-) 

It is available for download over GitHub. I still have unused TH-PCBs of V1 lying around, so currently I'm not too eager to spend time and money on testing v2.

**Reply by stevetuc** [reply]
![20250801_104632](https://github.com/user-attachments/assets/7feca422-db4c-42d1-a8b3-2e9d6b9c7eeb)

I built the v2 beta with Oled and seems to be working fine.

**Reply by glaszczkowski** [reply]
> I built the v2 beta with Oled and seems to be working fine.

Great! I have ordered the pcbs to test but still waiting for delivery. Good to know it works!

**Comment by stevetuc** [comment]
Are there any future plans to add 4116 support( a la ZX Spectrum)? The user would have to supply external -5 and 12 volts on pins 1& 8.  Maybe analog inputs could be used to sense these are correctly applied ( via resistor dividers) and control testing with pause.

**Reply by tops4u** [reply]
To be Honest, I planed this Tester with larger RAM in Mind - initially 44256 and 514000 but then extended it to other types as far as the design seemed fit to use them. However the 4116 is rather special. I order to really use it I'd rather create a intermediate Socket to accommodate it and generate the needed Voltages. I'll be thinking about it - on the other hand there are many 4116 Testers around already. I'll check the design in the next days. Could you open an Issue for this feature request? 

Regarding the v2 TH PCB. Thanks a lot for testing it, glad it works. I'll be replacing the V1 design on PCBWay and here on Github with this design. Could you send a Picture of you build in this Discussion?

**Reply by stevetuc** [reply]
Thanks for considering this. It would be very useful to have a single tester that handled all the vintage ramchips includiing those in the Spectrum :)
I have opened a feature request : https://github.com/tops4u/Ram-Tester/issues/7
I've added a pic of the v2 build in my earlier post.

**Reply by stevetuc** [reply]
Thanks for completing the 4116 adapter. Very cool! 
Is it possible to buy just the partially assembled adapter kit from amibay?
As already have the self assembled TH tester.

**Reply by tops4u** [reply]
Yes I think that will be possible. Do you want one of the protoypes or you wait for the more final version? Just PM me on Amibay so we can discuss how to handle this.

**Reply by stevetuc** [reply]
Thanks for posting the v33 4116 adapter schematic and pcb files. Do you think the design is fine as is , or do you plan to post an update with more esd protection at some point?
I preordered all the components but can wait to get pcb made if update is likely, as i dont have any 4116 need atm.

**Reply by tops4u** [reply]
I have a design ready with more ESD Protection for the DCDC but I doubt it is really necessary. The most ESD sensitive Part is the RAM itself, so you would need to have a ESD controlled environment to work with. There is also some EMI measures, but again, not sure it really is worth it. I can post the latest design, it is verified with one prototype. I would work, but you need a few more components. I would not post it in the repo at this time, but I can attach it here.

**Reply by stevetuc** [reply]
Thanks Andreas, that sounds good.

**Reply by stevetuc** [reply]
Im ready to build the 4116 board, so if you are able to post the latest design files here when convenient  that would be great.

**Reply by tops4u** [reply]
@stevetuc sorry it somehow got lost. Just updated to the latest Gerbers and added the BOM in the Readme. If you want to get the PCB all the parts and assembly service you'll it will close the price I ask.

**Comment by briantw** [comment]
Great design, Andreas! I covered it on [Elektor Magazine’s website](https://www.elektormagazine.com/news/open-source-diy-ram-tester) and there has been some interest, including a question about the BOM in the comments section.

**Reply by tops4u** [reply]
Hey there... thanks for the article. I just replied and hopefully answered this question.

**Comment by stevetuc** [comment]
Hi Andreas, Id like to try building myself for fun :) the mixed SMD/TH version of the pcb. Is V3 Gerber available or do you think v2.1 will be fine to build?

**Comment by Fuegomp3** [comment]
Hello, I'm trying to assemble the 4116 adapter, but I can't find the part numbers for the components. I don't know what each component is. Is there a location reference for each component in version 3.4?

Thanks in advance.

Nelson

**Reply by tops4u** [reply]
I'm currently abroad and habe limited access to the Internet. Please remind me mid may if I forgot to come back to you.

**Reply by Fuegomp3** [reply]
ok, thank you!

**Reply by Fuegomp3** [reply]
> I'm currently abroad and habe limited access to the Internet. Please remind me mid may if I forgot to come back to you.

Here I reminding you!

**Reply by tops4u** [reply]
Something like this?

<img width="625" height="727" alt="image" src="https://github.com/user-attachments/assets/23f07329-3817-421e-b1d0-f430a59f2b42" />

**Reply by Fuegomp3** [reply]
> Something like this?
> 
> <img alt="image" width="625" height="727" src="https://private-user-images.githubusercontent.com/6877067/593601283-23f07329-3817-421e-b1d0-f430a59f2b42.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzkyMzk0MzQsIm5iZiI6MTc3OTIzOTEzNCwicGF0aCI6Ii82ODc3MDY3LzU5MzYwMTI4My0yM2YwNzMyOS0zODE3LTQyMWUtYjFkMC1mNDMwYTU5ZjJiNDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDUyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjA1MjBUMDEwNTM0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDE1N2E3MDMxYTA4YjI1YTgxNGFlZGUxMjAzOGRmZDljN2FkMzRlZjg4MjNiMGM1NzVlNGUyMTExNWM4YmY4NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmcmVzcG9uc2UtY29udGVudC10eXBlPWltYWdlJTJGcG5nIn0.V54w3Snau4KJH63FdGJlzdtnkZB7w9NJ0daSixnsEoA">

Thank you!!

**Comment by ebastlir** [comment]
Hi, I'm not very experienced in the world of Arduino. I would like to change the DIP switch to the opposite. Now it's 16 18 20. I would like 20 18 16. I would like to edit the program, but I can't compile it. The compilation ends with an error for occupying 101% of the space. How to compile it successfully?

**Reply by tops4u** [reply]
You can not just change the Software as the DIP switches are hardwired to supply the chip with 5V. Otherwise you also need to adapt the schematic. Did you?

**Reply by ebastlir** [reply]
I have changed the schema.
I still can't translate, I have the U8g2 library installed

Arduino\libraries\U8g2\src/MUIU8g2.h:56:16: error: 'U8G2' has not been declared

**Reply by tops4u** [reply]
You would need to send me the whole source code. Please attach it as ZIP here.

**Comment by crtified-retro** [comment]
Is component C8 on the rear side of the 4116 board intended to be populated, and if Yes, what are the required specifications?

**Reply by crtified-retro** [reply]
Guessing by its location that a 16V or 25V 100nF 0603 or 0805 decoupling cap may be appropriate here, but be good to get official confirmation and possible BOM addition.

**Reply by tops4u** [reply]
Yes this is a 100nF 0603 type rated 16V is sufficient. It is for the RAMs 5V Line.

**Comment by crtified-retro** [comment]
I made a simple snap-in case-thing for my build of the main DRAM Tester board. It is just an under-tray. Will make the file available asap. First I need to minor edit, to make the clip thinner. As board insertion is currently difficult with stiff plastic. 

<img width="1558" height="1215" alt="rt_case_design_revA_cad" src="https://github.com/user-attachments/assets/ee6de6b1-462e-4066-b0b0-e0a802cc645a" />

**Reply by tops4u** [reply]
I just finally got my new 3D Printer and was also thinking about constructing a 3d Case. I'll try it if you want to send the STL to me. You may use contact@ramtester.ch

**Reply by crtified-retro** [reply]
I pushed back the top part of the clip by 0.5mm, in an attempt to make the board insertion possible and easier. 

I have just sent you the STL.

**Reply by tops4u** [reply]
Thanks for the STL. I think the PCB Cutouts are a bit too close for the PCB to fit in - at least for my printer it did not.

**Comment by attila-v** [comment]
Hi, thank you for your project, it is really useful!
I built it into a PCB breadboard, I made some modifications, you can see it here: https://avesoft.hu/ramtester.html
thx again,
Attila

**Comment by richardhopton** [comment]
Hi - I bought this from you a few months ago but it seems to have an older version of the firmware. I'm trying to follow your guidance but in Arduino IDE I get the error:
`Sketch uses 32308 bytes (100%) of program storage space. Maximum is 32256 bytes.`

and avrdude with an Arduino as ISP gives the error:
```
Warning: flash verification mismatch
  device 0xff != input 0x06 at addr 0x7800 (error)
Error: flash verification mismatch
```

**Reply by crtified-retro** [reply]
It may be this :
https://github.com/tops4u/Ram-Tester/issues/23

**Reply by attila-v** [reply]
a little workaround till issue will be solved is that comment out the 18 pin function caller in the Ram_Tester_5_0_7.ino at row 365-368 in the Arduino IDE - of course you can not check rare 4464 in this way, but the flash will be enough for other functions

**Reply by richardhopton** [reply]
Okay - I tried 5.0.6 with the arduinoisp and it appears to flash (no errors) and now the board runs 5.0.6!

**Reply by tops4u** [reply]
Hi there, sorry for the late reply. Yes 5.0.8 will probably be skipped, since there are a few other minor thinigs so 5.0.9 is in the works and will most probably be released in a few days. It will then also fit in the 328 including the Bootloader from the Arduino IDE - even if it is not needed. 
5.0.6 will work as well the fixes in 5.0.7 are only needed for some different OLED Displays in DIY TH Boards which did not support the now faster I2C Communication. So for SMD Users there is virtually no change except as slightly slower Test (due to slower Com with the Display).

**Comment by attila-v** [comment]
(
I forgot to remove CH341 from NANO, so VCC check fails with pin6 and pin7; workaround for me in 'common.cpp' at row 802:
`    if (bit & 0xfc) {
       if (md & bit) hd |= probeBit(&PORTD, &DDRD, &PIND, bit);
     }
`
)

**Reply by attila-v** [reply]
hm, I think the workaround is not good solution, because probably the Rx, Tx pins of the CH341 hold these address pins in high state, so the test can not check all cells of the RAM (and some cells checks more times)
I must to repair my device, thanks for your new VccProbe() feature!

**Reply by tops4u** [reply]
To be honest I have never updated the Tester using a CH341 which is not a method described in the Docs. However it could be useful to do it this way but for a proper operation the pins must be disconnected afterwards.
